packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -226,7 +226,19 @@ export class Span implements APISpan, ReadableSpan {
 
   setStatus(status: SpanStatus): this {
     if (this._isSpanEnded()) return this;
-    this.status = status;
+    this.status = { ...status };
+
+    // When using try-catch, the caught "error" is of type `any`. When then assigning `any` to `status.message`,
+    // TypeScript will not error. While this can happen during use of any API, it is more common on Span#setStatus()
+    // as it's likely used in a catch-block. Therefore, we validate if `status.message` is actually a string, null, or
+    // undefined to avoid an incorrect type causing issues downstream.
+    if (this.status.message != null && typeof status.message !== 'string') {
+      diag.warn(
+        `Dropping invalid status.message of type '${typeof status.message}', expected 'string'`
+      );
+      delete this.status.message;
+    }
```

Comment:
Is dropping the right behavior? If something is not nullish and also not a string, it may still be useful. Also, the user may be surprised to see nothing here. Maybe casting to string would be better?

Reply:
I've been going back and forth on this as well - I've ended up with this approach as dropping it here gives us the opportunity to add casting as a `string` later as a feature. 

I'm thinking that if we can get away with not doing any magic - then that's better. It will keep complexity lower, but also we discourage users from breaking the type-contract with the API.

Reply:
>I'm thinking that if we can get away with not doing any magic - then that's better. It will keep complexity lower, but also we discourage users from breaking the type-contract with the API.

I have to admit that I though about doing "magic" but I agree with you. If the API does magic I worry the users would ask for more of it.